joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.h

```diff
@@ -246,6 +252,54 @@ class JointTrajectoryController : public controller_interface::Controller<Hardwa
    */
   void setHoldPosition(const ros::Time& time, RealtimeGoalHandlePtr gh=RealtimeGoalHandlePtr());
 
+protected:
+  /**
+   * @returns the number of joints of the robot.
+   */
+  unsigned int getNumberOfJoints() const;
+
+  /**
+   * @brief Updates the states by sampling the specified trajectory for each joint
+   * at the specified sampling time.
+   *
+   * The current state is updated based on the values transmitted by the
+   * corresponding JointHandles.
+   *
+   * @param sample_time Time point at which the joint trajectories have to be sampled.
+   * @param traj Trajectory containing all joint trajectories currently under execution.
+   *
+   * @note This function is NOT thread safe but intended to be used in the
+   * update-function.
+   */
+  void updateStates(const ros::Time& sample_time, const Trajectory* const traj);
+
+protected:
+  /**
+   * @returns a trajectory consisting of joint trajectories with one pre-allocated
+   * segment.
+   */
+  static TrajectoryPtr createHoldTrajectory(const unsigned int& number_of_joints);
+
+private:
+  /**
+   * @brief Allows derived classes to perform additional checks
+   * and to e.g. replace the newly calculated desired value before
+   * the hardware interface is finally updated.
+   *
+   * @param time_data Updated time data.
+   */
+  virtual void updateFuncExtensionPoint(const Trajectory& curr_traj, const TimeData& time_data);
```

Comment:
Docu: Explain the arguments in more detail and state the unit's if not obvious.

Reply:
Done.